src/pages/item/item-actions/itemActionsButtons/loadRelationsButton/loadRelationsButton.tsx

```diff
@@ -0,0 +1,175 @@
+        const amountOfRelationsOnBoard =
+          await calculateAmountOfRelationsOnBoard(boardData);
+
+        setButtonDisabled(amountOfRelationsOnBoard == 0);
+        setDownstreamRefs(downstreamRefs);
+        setAssociations(associations);
+        setMiroBoardData(boardData);
+      }
+      setRelationsLoading(false);
+    }
+
+    fetchData();
+  }, [data]);
+
+  const calculateAmountOfRelationsOnBoard = async (boardData: BoardNode[]) => {
+    let count = 0;
+    await Promise.all(
+      data.outgoingAssociations.map(async function (outgoingAssociation) {
+        const appCardIds = await getAppCardIds(
+          outgoingAssociation.itemRevision.id,
+          boardData
+        );
+        count += appCardIds.length;
+      })
+    );
+    await Promise.all(
+      data.downstreamReferences.map(async function (downstreamReference) {
+        const appCardIds = await getAppCardIds(
+          downstreamReference.itemRevision.id,
+          boardData
+        );
+        count += appCardIds.length;
+      })
+    );
+    setRelationsOnBoardCount(count);
+    return count;
+  };
+
+  const onClickShow = async () => {
+    if (data && !buttonDisabled) {
+      const startCardIds = await getAppCardIds(
+        parseInt(props.itemId.toString()),
+        miroBoardData
+      );
+      const startCardId = startCardIds[0];
+      await createConnectorsForDownstreamRefsAndAssociation(
+        startCardId,
+        downstreamRefs,
+        associations,
+        miroBoardData
+      );
+    } else {
+      console.warn(
+        "Can't load Associations - data still loading or failed to do so."
+      );
+    }
+
+    setConnectorsAlreadyExist(!connectorsAlreadyExist);
+    const newBoard = await miro.board.get();
+    setMiroBoardData(newBoard);
+  };
+
+  const onClickHide = async () => {
+    if (data && !buttonDisabled) {
+      const startCardIds = await getAppCardIds(
+        parseInt(props.itemId.toString()),
+        miroBoardData
+      );
+      const startCardId = startCardIds[0];
+      await removeConnectors(startCardId, miroBoardData);
+    } else {
+      console.warn(
+        "Can't load Associations - data still loading or failed to do so."
+      );
+    }
+
+    setConnectorsAlreadyExist(!connectorsAlreadyExist);
+    const newBoard = await miro.board.get();
+    setMiroBoardData(newBoard);
+  };
+
+  return (
+    <>
+      <button
+        className={`button button-tertiary ${
+          relationsLoading ? "button-loading button-loading-primary" : ""
+        }`}
+        onClick={
+          connectorsAlreadyExist ? () => onClickHide() : () => onClickShow()
+        }
+        disabled={buttonDisabled}
+        data-test="show-dependency"
+        title={
+          connectorsAlreadyExist
+            ? `Hide Dependency & Associations (${relationsOnBoardCount})`
+            : `Show Dependency & Associations (${relationsOnBoardCount})`
+        }
```

Comment:
> [...] Icon-only buttons, with more specific icons & tooltip-on-hover [...]
> 
> Something along the line of this for example..
> 
> ![image](https://user-images.githubusercontent.com/70019423/226279143-95af9d37-7186-42e2-a166-c4b282f32df7.png)
> 

The `title` attribute and its rendering weren't what I meant by `tooltip` (which I tried to erase any doubt on by including it in the mockup, where the _tooltip_ I put in is explicitly an implementation of the [Material Design](https://m2.material.io/components/tooltips).)  
I thought that was clear and would ask you to use a proper tooltip implementation here. There's no shortage of libraries that can provide that, or you can create (y)our own.

Reply:
Tooltips look neat and implementation just so 👍🏽